x-pack/plugins/stack_alerts/public/alert_types/es_query/expression/search_source_expression_form.tsx

```diff
@@ -201,95 +203,64 @@ export const SearchSourceExpressionForm = (props: SearchSourceExpressionFormProp
       <EuiSpacer size="s" />
 
       <DataViewSelectPopover
-        initialDataViewTitle={dataView.title}
-        initialDataViewId={dataView.id}
+        initialDataViewTitle={dataView?.title}
+        initialDataViewId={dataView?.id}
         onSelectDataView={onSelectDataView}
       />
 
-      <EuiSpacer size="s" />
-
-      <SearchBar
-        onQuerySubmit={onQueryBarSubmit}
-        onQueryChange={onChangeQuery}
-        suggestionsSize="s"
-        displayStyle="inPage"
-        placeholder={i18n.translate('xpack.stackAlerts.searchSource.ui.searchQuery', {
-          defaultMessage: 'Search query',
-        })}
-        query={query}
-        indexPatterns={dataViews}
-        savedQuery={savedQuery}
-        filters={filters}
-        onFiltersUpdated={onUpdateFilters}
-        onClearSavedQuery={onClearSavedQuery}
-        onSavedQueryUpdated={onSavedQuery}
-        onSaved={onSavedQuery}
-        showSaveQuery={true}
-        showQueryBar={true}
-        showQueryInput={true}
-        showFilterBar={true}
-        showDatePicker={false}
-        showAutoRefreshOnly={false}
-        showSubmitButton={false}
-        dateRangeFrom={undefined}
-        dateRangeTo={undefined}
-        timeHistory={timeHistory}
-        hiddenFilterPanelOptions={['pinFilter', 'disableFilter']}
-      />
+      {dataViews.length > 0 && (
```

Comment:
In the case there were no data views available I think it wouldn't work creating this kind of alert rule. So we could check this earlier and display a message instead of the form

Reply:
This would be an empty view:

<img width="739" alt="Screenshot 2022-06-27 at 14 14 21" src="https://user-images.githubusercontent.com/1415710/175938725-d8a29262-65c3-4145-b084-2ca072bb586f.png">

Reply:
The question here is, are there any problems opening a flyout in a flyout?

Reply:
@andreadelrio What do you think about opening a flyout from a flyout? This way we could allow users to continue configuring their alert after they created a data view.

![Jun-27-2022 15-36-28](https://user-images.githubusercontent.com/1415710/175955436-ac6c81df-db28-4773-a24c-36607e879899.gif)

Reply:
looking at the gif, it LGTM

Reply:
I think ideally we would keep EuiExpression on the outside. Inside you can still include "Create data view" in the dropdown.

Reply:
@andreadelrio Sure, we can do that as well. Something like this?

<img width="590" alt="Screenshot 2022-06-28 at 10 16 51" src="https://user-images.githubusercontent.com/1415710/176129785-5ec0b2ef-63cd-4a81-856e-af7e2f069167.png">